android/src/main/java/com/amplitude/android/internal/fragments/FragmentActivityHandler.kt

```diff
@@ -0,0 +1,34 @@
+package com.amplitude.android.internal.fragments
+
+import android.app.Activity
+import android.os.Build
+import androidx.annotation.RequiresApi
+import androidx.fragment.app.FragmentActivity
+import com.amplitude.common.Logger
+import java.util.WeakHashMap
+
+internal typealias TrackEventCallback = (String, Map<String, Any?>) -> Unit
+
+@RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+internal object FragmentActivityHandler {
+    private val callbacksMap =
+        WeakHashMap<FragmentActivity, MutableList<AutocaptureFragmentLifecycleCallbacks>>()
+
+    fun Activity.registerFragmentLifecycleCallbacks(track: TrackEventCallback, logger: Logger) {
+        (this as? FragmentActivity)?.apply {
```

Comment:
Could you handle the case where type cast from Activity to FragmentActivity fails? 

If type cast fails, what will happen?

Reply:
If the cast fails, the activity is not a `FragmentActivity` and does not contain any fragments.

Reply:
Please let me know if there is any specific behavior that we want for the failure case.

Reply:
I was thinking if we should track when type cast fails or at least log out for debugging, for example, fragment tracking is not working because the type cast failure.